pkg/xds/topology/outbound.go

```diff
@@ -180,17 +182,15 @@ func fillRemoteMeshServices(
 	zoneIngress []*core_mesh.ZoneIngressResource,
 	mesh *core_mesh.MeshResource,
 	localZone string,
+	mtlsEnabled bool,
 ) {
-	ziInstances := map[string]struct{}{}
-
-	if !mesh.MTLSEnabled() {
-		// Ingress routes the request by TLS SNI, therefore for cross
-		// cluster communication MTLS is required.
-		// We ignore Ingress from endpoints if MTLS is disabled, otherwise
-		// we would fail anyway.
+	if !mtlsEnabled {
```

Comment:
Since we might have meshidentity instead of mTLS on mesh we need to check both (in mesh context is a helper function)